micrometer-api/src/main/java/io/micrometer/api/instrument/observation/ObservationRegistry.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2017 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.api.instrument.observation;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.function.BiPredicate;
+
+import io.micrometer.api.instrument.MeterRegistry;
+import io.micrometer.api.instrument.NoopObservation;
+import io.micrometer.api.lang.Nullable;
+
+/**
+ */
+public interface ObservationRegistry {
+    @Nullable
+    Observation getCurrentObservation();
+
+    void setCurrentObservation(@Nullable Observation current);
+
+    /**
+     * !!!!!!!!!!!!!!!!!!!! THIS IS NOT STARTED !!!!!!!!!!!!!!!!!!!!
+     * !!!!!!!!!!!!!!!!!!!! REMEMBER TO CALL START() OTHERWISE YOU WILL FILE ISSUES THAT STUFF IS NOT WORKING !!!!!!!!!!!!!!!!!!!!
+     * @param name observation name
+     * @return this
+     */
+    default Observation observation(String name) {
+        if (config().isObservationEnabled(name, null)) {
+            return new SimpleObservation(name, this, new Observation.Context());
+        }
+        return NoopObservation.INSTANCE;
+    }
+
+    /**
+     * !!!!!!!!!!!!!!!!!!!! THIS IS NOT STARTED !!!!!!!!!!!!!!!!!!!!
+     * !!!!!!!!!!!!!!!!!!!! REMEMBER TO CALL START() OTHERWISE YOU WILL FILE ISSUES THAT STUFF IS NOT WORKING !!!!!!!!!!!!!!!!!!!!
+     * @param name observation name
+     * @param context context
+     * @return this
+     */
+    default Observation observation(String name, Observation.Context context) {
+        if (config().isObservationEnabled(name, context)) {
+            return new SimpleObservation(name, this, context);
+        }
+        return NoopObservation.INSTANCE;
+    }
+
+    /**
+     * Started observation.
+     *
+     * @param name observation name
+     * @return this
+     */
+    default Observation start(String name) {
+        return observation(name).start();
+    }
+
+    /**
+     * Started observation.
+     *
+     * @param name observation name
+     * @param context context
+     * @return this
+     */
+    default Observation start(String name, Observation.Context context) {
+        return observation(name, context).start();
+    }
+
+    Config config();
+
+    /**
+     * Access to configuration options for this registry.
+     */
+    class Config {
```

Comment:
Add configuration option for asserting whether observation was started when stop, etc. Choose a strategy (throw exception, log error, warning)